### ISSUE TITLE
basic auth dialog

### ISSUE THREAD

**Issue by mderazon** [puppeteer/puppeteer#418]
Can't find a way to fill the user/password in the basic authentication dialog in the browser:

<img width="917" alt="screen shot 2017-08-20 at 15 14 33" src="https://user-images.githubusercontent.com/717076/29494796-6a37667e-85ba-11e7-957a-26c0d50a4f99.png">

Any idea ?

**Comment by mderazon** [comment]
Okay that didn't take too long to figure out.

You need to use `Authorization` header:
``` js
const puppeteer = require('puppeteer');

const user = 'myuser';
const pass = 'mypass';

const headers = new Map();
headers.set(
  'Authorization',
  `Basic ${new Buffer(`${myuser}:${mypass}`).toString('base64')}`
);

(async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  await page.setExtraHTTPHeaders(headers);
  await page.goto('https://example.com');
  
  // ...
})();

```

Could have been more elegant, but works...

**Comment by aslushnikov** [comment]
For the reference: a more elegant approach is suggested in #426

**Comment by Aaronphy** [comment]
@aslushnikov  when I use , it shows "page.authenticate is not a function"

**Comment by pcj** [comment]
For me, it did not work with Map, :

```typescript
const auth = new Buffer(`${user}:${pass}`).toString('base64');
await page.setExtraHTTPHeaders({
    'Authorization': `Basic ${auth}`                    
});
page.on('request', request => console.log(`Request: ${request.resourceType}: ${request.url} (${JSON.stringify(request.headers)})`));

```

**Comment by konstantinblaesi** [comment]
@Aaronphy page.authenticate(credentials) is only available since 0db6165d7389df6f1a3a4c09fcf1d338bc3351a2 . There was no new release since this was added. You could install the latest master with
`npm i --save https://github.com/GoogleChrome/puppeteer.git`
or wait for puppeteer > v0.10.2

**Comment by cassegfault** [comment]
@konstantinblaesi Thank you, have been trying to get this to work all morning and couldn't figure out why it wasn't working... It may be nice to have `Since vX.Y` on the docs
